# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/nsAccUtils.cpp
# Commit: 09de143215a1
# Full Hash: 09de143215a13a22dcb2cd53d4ed1dfb60480c82
# Author: Atila Butkovits <abutkovits@mozilla.com>
# Date: 2025-09-30 02:48 +0300
# Regressor Bug: 1987945
# File Overlap Count: 2
# Overlapping Files: accessible/base/nsCoreUtils.cpp, accessible/tests/browser/relations/browser_anchor_positioning.js
# Description:
#   Revert "Bug 1987945: apply code formatting via Lando" for causing bustages at DocAccessible.cpp.
# ==============================================================================

diff -r 826eb4ac00dd -r 09de143215a1 accessible/base/nsAccUtils.cpp
--- a/accessible/base/nsAccUtils.cpp	Tue Sep 30 02:35:22 2025 +0300
+++ b/accessible/base/nsAccUtils.cpp	Tue Sep 30 02:48:50 2025 +0300
@@ -651,29 +651,3 @@
   return aAccessible->IsTextField() ||
          aAccessible->Elm()->State().HasState(dom::ElementState::READWRITE);
 }
-
-bool nsAccUtils::IsValidDetailsTargetForAnchor(const Accessible* aTarget,
-                                               const Accessible* aAnchor) {
-  if (aAnchor->IsAncestorOf(aTarget)) {
-    // If the anchor is a parent of the target, the target is not valid
-    // relation.
-    return false;
-  }
-
-  Accessible* nextSibling = aAnchor->NextSibling();
-  if (nextSibling && nextSibling->IsTextLeaf()) {
-    nsAutoString text;
-    nextSibling->Name(text);
-    if (nsCoreUtils::IsWhitespaceString(text)) {
-      nextSibling = nextSibling->NextSibling();
-    }
-  }
-
-  if (nextSibling == aTarget) {
-    // If the target is the next sibling of the anchor (ignoring whitespace
-    // text nodes), the target is not a valid relation.
-    return false;
-  }
-
-  return true;
-}